build_defs/python.build_defs

```diff
@@ -196,6 +202,18 @@ def python_binary(name:str, main:str, srcs:list=[], resources:list=[], out:str=N
     )
 
 
+def _set_shebang(interpreter:str, interpreter_options:str, site:bool):
+    """Returns a shebang to pass to arcat to inject at the top of a pex"""
+    # Assume the path to the interpreter is relative
+    shebang = f'/usr/bin/env {interpreter}'
+    if not site:
+        shebang = f'#!/bin/sh\nexec {shebang} -S \\\\$0 \\\"\\\\$@\\\"'
+    if not shebang.startswith('#!'):
+        shebang = '#!' + shebang
```

Comment:
Can't this just be
```python
if site:
    return f'#!/usr/bin/env {interpreter}'
else: 
    return f'#!/bin/sh\nexec {shebang} -S \\\\$0 \\\"\\\\$@\\\"'
```

Also, is there a reason this need to change so much? The old shebang code was just:
```
    shebang = shebang or interpreter or CONFIG.PYTHON.DEFAULT_INTERPRETER
```

I think the pex tool does something very similar to this, so maybe we just need to tell it which python interpreter we're using? Looks like it might just be adding that to the beginning of the pex main `.py` file here rather than the actual `.pex`?

Reply:
Yeah it could definitely be the top one. I can't remember exactly but I think the idea behind changing this was that, like you said below, we're not doing any shebang logic in the pex tool anymore and we want to do it here instead.